src/components/VerticalNavigation/VerticalNavigationItem.js

```diff
@@ -0,0 +1,443 @@
+      }
+    } else {
+      this.setState({ [stateKey]: !pinned });
+      setActivePath(!pinned ? this.idPath() : null);
+      if (pinned) {
+        forceHideSecondaryMenu();
+        this.onItemBlur(true);
+      }
+    }
+    updateNav(!pinned);
+  }
+
+  pinSecondaryNav() {
+    // TODO these don't really need to be separate by depth, do they...?
+    this.pinNav('secondaryPinned', this.props.updateNavOnPinSecondary);
+  }
+
+  pinTertiaryNav() {
+    this.pinNav('tertiaryPinned', this.props.updateNavOnPinTertiary);
+  }
+
+  onItemHover() {
+    const { primary, secondary, tertiary } = this.getContextNavItems();
+    const {
+      isMobile,
+      hoverDelay,
+      updateNavOnItemHover,
+      onHover,
+      setControlledState,
+      hovering
+    } = this.props;
+    const { hoverTimer } = this.state;
+    const that = this;
+    const item = deepestOf(primary, secondary, tertiary);
+    if (item.subItems && item.subItems.length > 0) {
+      if (!isMobile) {
+        if (hoverTimer) {
+          clearTimeout(hoverTimer);
+          this.setState({ hoverTimer: null });
+        } else if (!hovering) {
+          this.setState({
+            hoverTimer: setTimeout(() => {
+              that.setState({ hoverTimer: null });
+              setControlledState({ hovering: true });
+              updateNavOnItemHover(primary, secondary, tertiary);
+              onHover && onHover(primary, secondary, tertiary);
+            }, hoverDelay)
+          });
+        }
+      }
+    }
+  }
+
+  onItemBlur(immediate) {
+    const { primary, secondary, tertiary } = this.getContextNavItems();
+    const {
+      hideDelay,
+      updateNavOnItemBlur,
+      onBlur,
+      setControlledState,
+      hovering
+    } = this.props;
+    const { hoverTimer } = this.state;
+    const that = this;
+    const item = deepestOf(primary, secondary, tertiary);
+    if (item.subItems && item.subItems.length > 0) {
+      if (hoverTimer) {
+        clearTimeout(hoverTimer);
+        this.setState({ hoverTimer: null });
+      } else if (hovering) {
+        this.setState({
+          hoverTimer: setTimeout(() => {
+            that.setState({ hoverTimer: null });
+            setControlledState({ hovering: false });
+            updateNavOnItemBlur(primary, secondary, tertiary);
+            onBlur && onBlur(primary, secondary, tertiary);
+          }, immediate ? 1 : hideDelay)
+        });
+      }
+    }
+  }
+
+  onItemClick() {
+    const { primary, secondary, tertiary } = this.getContextNavItems();
+    const { isMobile, updateNavOnItemClick, onClick } = this.props;
+    updateNavOnItemClick(primary, secondary, tertiary); // Clears all mobile selections
+    if (isMobile) {
+      this.onMobileSelection(primary, secondary, tertiary); // Applies new mobile selection here
+    }
+    this.setActive();
+    onClick && onClick(primary, secondary, tertiary);
+    // TODO other item-level nav props? href? route?
```

Comment:
Selecting a leaf node should dismiss the menus.

Reply:
Ah, good to know.